static-analysis/teammates-macker.xml

```diff
@@ -1,45 +1,340 @@
+                            <include class="teammates.ui.template.AdminAccountDetailsInstructorCourseListTableRow" />
+                            <include class="teammates.ui.template.AdminAccountDetailsStudentCourseListTableRow" />
+                            <include class="teammates.ui.template.AdminAccountManagementAccountTableRow" />
+                            <include class="teammates.ui.template.FeedbackSessionPublishButton" />
+                            <include class="teammates.ui.template.FeedbackSessionsAdditionalSettingsFormSegment" />
+                            <include class="teammates.ui.template.FeedbackSessionsForm" />
+                            <include class="teammates.ui.template.InstructorFeedbackSessionActions" />
+                            <include class="teammates.ui.template.StudentProfileEditBox" />
+                        </from>
+                    </allow>          
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Controller should be self-contained">
+            <access-rule>
+                <deny>
+                    <from pattern="productionCode" />
+                    <to class="teammates.ui.controller.**" />
+                    <allow>
+                        <from class="teammates.ui.controller.**" />
+                    </allow>          
+                </deny>
+            </access-rule>
+        </ruleset>
+        
+        <ruleset name="LogicComponent: Each logic can only access its corresponding DB (e.g. AccountsLogic -> AccountsDb)">
+            <foreach var="logicModule" class="teammates.logic.core.(*)Logic">
+                <pattern name="DbModule" class="teammates.storage.api.${logicModule}Db" />
+                
+                <access-rule>  
+                    <message>${logicModule} can only access its corresponding DB</message>
+                    <deny> 
+                        <from class="teammates.logic.core.${logicModule}Logic" />
+                        <to class="teammates.storage.api.**" /> 
+                        <allow>
+                            <to pattern="DbModule" />
+                        </allow>
+                    </deny>
+                </access-rule>
+            </foreach>
+        </ruleset>
+        <ruleset name="LogicComponent: BackDoorServlet should not access logic:core">
+            <access-rule>
+                <deny>
+                    <from class="teammates.logic.backdoor.BackDoorServlet" />
+                    <to class="teammates.logic.core.**" />      
+                </deny>
+            </access-rule>
+        </ruleset>
+        
+        <ruleset name="StorageComponent: storage::entity should only be accessed by storage::api and common:datatransfer">
+           <access-rule>
+                <deny>
+                    <from pattern="productionCode" />
```

Comment:
Here as well; if the ruleset is about `StorageComponent`, it should simply be `teammates.storage.**`